hadoop/src/main/java/org/apache/carbondata/hadoop/util/CarbonInputFormatUtil.java

```diff
@@ -70,6 +69,18 @@ public static CarbonQueryPlan createQueryPlan(CarbonTable carbonTable, String co
           i++;
         }
       }
+    } else {
+      // query all columns
+      List<CarbonMeasure> tableMsrs = carbonTable.getMeasureByTableName(factTableName);
```

Comment:
This is the bug, when input projection is null, earlier it does not add all columns to read